bynder_sdk/oauth2.py

```diff
@@ -1,41 +1,125 @@
-import random
-import string
-
+from oauthlib.oauth2 import BackendApplicationClient
+from oauthlib.oauth2.rfc6749.errors import (
+    MissingTokenError,
+    TokenExpiredError,
+    OAuth2Error,
+)
 from requests_oauthlib import OAuth2Session
 
-from bynder_sdk.util import SessionMixin
+from bynder_sdk.version import VERSION
+
+UA_HEADER = {
+    'User-Agent': 'bynder-python-sdk/{}'.format(VERSION)
+}
+
 
+def api_endpoint_url(session, endpoint):
+    if endpoint.startswith('/v6') or endpoint.startswith('/v7'):
+        return 'https://{}{}'.format(session.domain, endpoint)
+    return 'https://{}/api{}'.format(session.domain, endpoint)
 
-def oauth2_url(bynder_domain, endpoint):
-    return 'https://{}/v6/authentication/oauth2/{}'.format(
-        bynder_domain, endpoint)
 
+def oauth2_url(domain, endpoint):
+    return 'https://{}/v6/authentication/oauth2/{}'.format(domain, endpoint)
 
-class BynderOAuth2Session(SessionMixin, OAuth2Session):
-    def __init__(self, bynder_domain, *args, **kwargs):
-        self.bynder_domain = bynder_domain
 
-        kwargs['auto_refresh_url'] = oauth2_url(self.bynder_domain, 'token')
+class BynderOAuth2Session(OAuth2Session):
+    # pylint: disable-msg=too-many-arguments
+    def __init__(
+        self,
+        domain,
+        client_id,
+        client_secret,
+        scope,
+        redirect_uri=None,
+        token=None,
+        token_updater=None,
+        **kwargs
```

Comment:
What other `kwargs` could we expect here?

Reply:
`OAuth2Session` accepts `kwargs` which are then passed on to `requests.Session`, I figured it would make sense to keep that in as well.